js/ui/grid_core/ui.grid_core.header_filter_core.js

```diff
@@ -288,7 +288,9 @@ exports.HeaderFilterView = modules.View.inherit({
                         const selectedItems = e.component.option('selectedItems');
 
                         if(!e.component._selectedItemsUpdating && !e.component.option('searchValue') && !options.isFilterBuilder) {
-                            if(selectedItems.length === 0 && items.length && (!options.filterValues || options.filterValues.length <= 1)) {
+                            const filterValues = options.filterValues || [];
+                            const isExclude = options.filterType === 'exclude';
+                            if(selectedItems.length === 0 && items.length && (filterValues.length <= 1 || isExclude && filterValues.length === items.length - 1)) {
```

Comment:
Maybe move this huge condition to the const with charactonym? 
Or at least this part
`(filterValues.length <= 1 || isExclude && filterValues.length === items.length - 1)`

Reply:
But how to name this constant? I think that the name will not be so obvious.